caput/mpiarray.py

```diff
@@ -89,6 +92,100 @@
             print(ri, arr3)
         mpiutil.barrier()
 
+Direct Slicing
+==============
+
+The :class: `MPIArray` supports direct slicing, with `__getitem__`.
+This can be used for both fetching and assignment. It is recommended
+to only index into the non-parallel axis or to do a full slice `[:]`.
+
+Behaviour
+
+- A full slice `[:]` will return a :class: `MPIArray` on fetching, with
+identical properties to the original array.
+- Any indexing or slicing into the non-parallel axis, will return a
+:class: `MPIArray`. The number associated with the parallel axis,
+will be adjusted if a slice results in an axis reduction.
+- Any indexing into the parallel axis will result into a local index on each rank,
+returning a regular `numpy` array.
+
+Example
+-------
+
+    import numpy as np
```

Comment:
I would suggest that you make this example runnable via doctest (see `caput.config` for an example). That way we'll get some bonus testing, but more importantly it means that the example will work and won't end up out of date. Practically that means putting `>>>` in front of some lines, putting a `True` under the test lines and you'll need to make it agnostic on the communicator size (as doctest will only run with a size=1), but that should be as easy as just replacing the 4 with `mpiutil.size`.